### DISCUSSION TITLE
dependencyMatcher.pipe

### DISCUSSION THREAD

**Discussion by theognis0110** [explosion/spaCy#7809] [Help: Best practices]
Hi,
Is there a way to process a batch of docs with the new dependencyMatcher as in the "ancient time" with matcher.pipe

i used to do 
        for i, (doc, matches) in enumerate(matcher.pipe(nlp.pipe(texts, batch_size=lenn),
                                             batch_size=lenn, return_matches=True)):
               do_something(doc, matches)

but this is no longer possible

**Comment by polm** [comment]
You can just call the matcher inside the for loop. The implementation of the old pipe function on the matcher is just a for loop anyway. 

```
for i, doc in enumerate(nlp.pipe(texts, batch_size=lenn)):
    matches = matcher(doc)
    do_something(doc, matches)
```

**Reply by theognis0110** [reply]
ok, the simpler the better
would it make sense to use Doc.from_docs (merge several docs into one) so as to increase overall performances (avoid repetitive calls in case of many docs) ?

**Reply by polm** [reply]
I don't think that'll make a difference. `nlp.pipe` is faster because it allocates resources to be reused, but the dependency matcher doesn't have setup/teardown like that.
